config/application.php

```diff
@@ -106,6 +106,7 @@
  * Debugging Settings
  */
 Config::define('WP_DEBUG_DISPLAY', false);
+Config::define('WP_DEBUG_LOG', env('WP_DEBUG_LOG') ?? $webroot_dir . Config::get('CONTENT_DIR') . '/uploads/debug.log');
```

Comment:
```php
WP_CONTENT_DIR . '/uploads/debug.log'
```

Reply:
`Config::apply` is not yet applied at this point.
`WP_CONTENT_DIR` is most likely undefined.

Reply:
Even with `Config::get('WP_CONTENT_DIR')`? 🤔

Reply:
```php
Config::define('WP_DEBUG_LOG', env('WP_DEBUG_LOG') ?? "${WP_CONTENT_DIR}/uploads/debug.log"); 

Config::apply();
==> PHP Notice:  Undefined variable: WP_CONTENT_DIR in ......./config/application.php on line 129

echo WP_DEBUG_LOG;
==> /uploads/debug.log
```

Reply:
can we avoid setting a default here?

Reply:
🤦‍♂ read the PR comments 😛 of course we can avoid it. the question is if a default is better than none

Reply:
~~Can we~~ I recommend avoiding setting a default here. This could cause unintended consequences on high-availability setups. In the case WP_DEBUG is set to true on purpose or on accident this is a fail-open behavior rather than fail-safe.

Reply:
> Even with Config::get('WP_CONTENT_DIR')? 

I think `Config::get` is possible.

---

How about we settle for:

```php
Config::define('WP_DEBUG_LOG', env('WP_DEBUG_LOG') ?? false); 
```

Reason: https://github.com/WordPress/WordPress/blob/7004afe4f4bac1fd17a142051832bdf6be8e6fcf/wp-includes/default-constants.php#L87-L89

Reply:
Yup that LGTM. fail-safe default.

Reply:
And, what do we do on development?

Seems difficult to get a path for when using valet

Reply:
In development the developer just does whatever works best for their development setup.